exchanges/binance/binance.go

```diff
@@ -1209,72 +1211,65 @@ func (b *Binance) MaintainWsAuthStreamKey(ctx context.Context) error {
 	}, request.AuthenticatedRequest)
 }
 
-// FetchSpotExchangeLimits fetches spot order execution limits
-func (b *Binance) FetchSpotExchangeLimits(ctx context.Context) ([]order.MinMaxLevel, error) {
-	var limits []order.MinMaxLevel
-	spot, err := b.GetExchangeInfo(ctx)
+// FetchExchangeLimits fetches order execution limits filtered by asset
+func (b *Binance) FetchExchangeLimits(ctx context.Context, a asset.Item) ([]order.MinMaxLevel, error) {
+	if a != asset.Spot && a != asset.Margin {
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, a)
+	}
+
+	resp, err := b.GetExchangeInfo(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	for x := range spot.Symbols {
+	limits := make([]order.MinMaxLevel, 0, len(resp.Symbols))
+	for _, s := range resp.Symbols {
 		var cp currency.Pair
-		cp, err = currency.NewPairFromStrings(spot.Symbols[x].BaseAsset,
-			spot.Symbols[x].QuoteAsset)
+		cp, err = currency.NewPairFromStrings(s.BaseAsset, s.QuoteAsset)
 		if err != nil {
 			return nil, err
 		}
-		var assets []asset.Item
-		for y := range spot.Symbols[x].Permissions {
-			switch spot.Symbols[x].Permissions[y] {
-			case "SPOT":
-				assets = append(assets, asset.Spot)
-			case "MARGIN":
-				assets = append(assets, asset.Margin)
-			default:
-				// "LEVERAGED", "TRD_GRP_003", "TRD_GRP_004", "TRD_GRP_005" etc are unused permissions
-				// for spot exchange limits
-			}
+
+		if !slices.Contains(s.Permissions, strings.ToUpper(a.String())) {
```

Comment:
Only thing I'd suggest here is to move `strings.ToUpper(...)` out of the `for` so you don't need to process it for every symbol and for each item in `s.Permissions`

Reply:
Agreed.

Fixed 1ee33f73